src/configurationProvider.ts

```diff
@@ -195,16 +194,115 @@ export class JavaDebugConfigurationProvider implements vscode.DebugConfiguration
         return !Array.isArray(configItems) || !configItems.length;
     }
 
-    private async chooseMainClass(folder: vscode.WorkspaceFolder | undefined): Promise<IMainClassOption | undefined> {
-        const res = await resolveMainClass(folder ? folder.uri : undefined);
+    private async quickFixLaunchConfig(folder: vscode.Uri | undefined, config: vscode.DebugConfiguration):
```

Comment:
I don't think this function is to fix the launch config; fix is only part of the functionality after failed to choose mainclass. More like populate the launchConfig.

Reply:
IMO, when the mainClass and projectName is empty, and we prompt users to choose one, this is also one kind of fix. And the difference is we don't persist them in launch.json.

Reply:
Fix means something is wrong and you want to correct it. For user configuration, the mainClass and projectName is correct and doesn't need fix them after validation.